toolchainlib/src/compression.luau

```diff
@@ -36,70 +33,51 @@ local function detectFormat(fileName: string): Option<CompressionFormat>
 end
 
 -- TODO: Use a type function to make all CompressionFormat lowercase
-local decompress: { [CompressionFormat]: (compressed: buffer) -> Result<pathfs.AsPath, string> } = {
-	Zip = function(compressed: buffer)
-		return (Option.from(dirs.cacheDir()):map(function(cacheDir)
-			local progCacheDir = cacheDir:join("pesde-bin")
-			if not pathfs.isDir(progCacheDir) then
-				pathfs.writeDir(progCacheDir)
+local extractBinary: {
+	[CompressionFormat]: (
+		compressed: buffer,
+		binaryName: string,
+		targetPlatform: PlatformDescriptor.PlatformDescriptor
+	) -> Result<buffer, string>,
+} =
+	{
+		Zip = function(compressed, binaryName, targetPlatform)
+			local reader = unzip.load(compressed)
+
+			local binaryEntry = reader:findEntry(binaryName)
+			local binaryContents = nil
```

Comment:
I would stylistically prefer it if we did not explicitly initialize the variable and added a type annotation to it instead.